predator/asp/greedy-target-seed-solving_SA.lp

```diff
@@ -46,32 +39,23 @@ present(M) :- any_seed(M).
 % % a metabolite is present or not
 % { present(M) } :- metabolite(M).
 
-% % products of a feasible reaction are producible
-% producible(M) :- feasible(R), product(M,R).
+% products of a feasible reaction are producible
+producible(M) :- feasible(R), product(M,R).
 
 % % every present metabolite must be producible
 % :- present(M), not producible(M), not any_seed(M).
 
 %%%%%%% version 2 %%%%%%%%
 
-% a metabolite is present if it is given as a seed or if it is the product of a feasible reaction
-present(X) :- product(X,R), feasible(R).
 
+% try to cut the set of possibly present metabolites
 reach(X) :- any_seed(X).
 reach(X) :- product(X,R), reactant(Y,R), reach(Y).
 
 % a metabolite might be present or not
 { present(X) } :- product(X,R), reach(X). 
 
-% atleast one reaction must be producing X
-is_produced(X) :- feasible(R), product(X,R).
-:- not is_produced(X), present(X), not any_seed(X).
-
-% there must be a path from a seed to the present metabolite
-founded(X) :- any_seed(X).
-founded(X) :- product(X,R), reactant(Y,R), founded(Y), feasible(R).
-
-:- present(X), not founded(X).
+:- present(X), not producible(X).
```

Comment:
Integrity constraint that ensures that every present metabolites is either producible or a seed. Just simpler.
Edit: I actually forgot the seeds. Corrected i must be: `:- present(X), not any_seed(X), not producible(X).` 
The next commit should fix that.